### ISSUE TITLE
Google Chart Library blocks parsing of document while page load.

### ISSUE THREAD

**Issue by parin1990** [primefaces-extensions/primefaces-extensions.github.com#452]
Google chart library is blocking parsing of DOM while it is being loaded as it is loaded via `document.write` from the component.As a result, the mobile browser rendering is blocked for couple of seconds until the library is loaded. In desktop browser console following warning is being thrown.

![image](https://cloud.githubusercontent.com/assets/4145068/25649995/70f0959e-2ff7-11e7-8401-fe3a70469457.png)

**Comment by melloware** [comment]
You are right.  It is doing this...

```java
		this.startScript(writer, chart.getClientId());
		writer.writeAttribute("src", "https://www.google.com/jsapi", null);
		this.endScript(writer);
```

I have to remove this code and looking at PrimeFaces GMap component they just require you to include it yourself on the page.  See:  https://www.primefaces.org/showcase/ui/data/gmap/basic.xhtml

So I will remove the code from inside the java and now on your page you will need to include it yourself like...

`<script src="https://www.google.com/jsapi"></script>`

I believe this will fix both #452 and #453 by doing this.

**Comment by parin1990** [comment]
True. the renderer is creating `<script/>` tags. also i found that `google` variable is being used in `gchart.js` and one another file.So i guess javascript widget files will also require a modifications.

**Comment by melloware** [comment]
actually i think it might be worse than that.  Look at this code..

https://www.google.com/jsapi

Google itself is loading scripts inside that script like this..

google.loader.ServiceBase = 'https://www.google.com/uds';

And if you notice the Chrome warning is saying "A Parser-blocking, cross site (i.e. different eTLD+1) script, https://www.google.com/uds/?file=visualization&v=1.0&packages=corechart" that Google UDS is causing the issue because its doing a doc write from right inside their own JS file.

I moved the <script> code out of the component and I am still getting this error.   I think moving the script tag out of the component will only solve the "loading the script only once" for the whole page rather than for each GChart on your screen.

**Comment by parin1990** [comment]
`https://www.google.com/jsapi` library is now upgraded with another library.

see https://developers.google.com/chart/interactive/docs/basic_load_libs

Yes you are right. library loading should be in `<head/>` section  of the page and not inside a component.

**Comment by melloware** [comment]
OK I just upgrade to use the new way of loading GCharts from your article above.  I have verified in my local showcase all of the browser warnings go away and the script is only loaded once.

I posted a SNAPSHOT of 6.1.1 if you want to give it shot...

https://oss.sonatype.org/content/repositories/snapshots/org/primefaces/extensions/primefaces-extensions/6.1.1-SNAPSHOT/

Note: you will need to add this library yourself to your JSF page....

`<script src="https://www.gstatic.com/charts/loader.js"></script>`

**Comment by parin1990** [comment]
Not working @melloware . actually you will need to change `gchart.js` as well.

**Comment by melloware** [comment]
I did change gchart.js and its working fine in my showcase locally.    Did that with this commit..

https://github.com/primefaces-extensions/core/commit/67caec71ae63d59187c7f003f15d6f6be47ec476

**Comment by melloware** [comment]
In 6.1.1 it gets merged into primefaces-extensions.js as a combined file that is why you don't see a gchart.js in 6.1.1.  You are on a WAY old version of 3.0.0 which used to have all the JS files separate.

**Comment by melloware** [comment]
Here is the raw file for you to view and copy...

https://github.com/primefaces-extensions/core/blob/master/src/main/resources/META-INF/resources/primefaces-extensions/gchart/gchart.js

**Comment by parin1990** [comment]
Great !. its working properly now.

**Comment by melloware** [comment]
Awesome!  I will mark this closed thanks for testing it.
